mautic/mautic
# [DPMMA-2592] Save selected template HTML without opening the builder

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [❌] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR adds the ability to save the HTML of the selected template based on MJML without the opening the builder. Previously, selecting a template without opening the builder did not save the styling in the message properly.

The "mjmlToHtml" method should return a string type, but I received an object, so the type of the variable is verified in the condition before assignment.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here]
(https://contribute.mautic.org/contributing-to-mautic/tester))
2. Make sure GrapesJS is enabled in Plugins. 
3. Just in case, rebuild the assets using the "php bin/console m:a:g" command.
4. Create a new email message and select one of the MJML templates.
5. Save the email without opening the GrapesJS builder.
6. Send the message to your test contacts. Email should be displayed correctly with styling.
